packages/clay-label/src/index.tsx

```diff
@@ -100,35 +107,46 @@ const ClayLabel = React.forwardRef<HTMLAnchorElement | HTMLSpanElement, IProps>(
 				})}
 				ref={ref}
 			>
-				{contentBefore && (
-					<ClayLabelItemBefore>{contentBefore}</ClayLabelItemBefore>
-				)}
-
-				<TagName
-					{...innerElementProps}
-					className={classNames(
-						'label-item label-item-expand',
-						innerElementProps.className
-					)}
-					href={href}
-				>
-					{children}
-				</TagName>
+				{children}
 
 				{closeButtonProps && (
-					<span className="label-item label-item-after">
+					<ClayLabelItemAfter>
 						<button
 							{...closeButtonProps}
-							className="close"
+							className={classNames(
+								closeButtonProps.className,
+								'close'
+							)}
 							type="button"
 						>
-							<Icon spritemap={spritemap} symbol="times" />
+							<ClayIcon spritemap={spritemap} symbol="times" />
 						</button>
-					</span>
+					</ClayLabelItemAfter>
 				)}
```

Comment:
Maybe move this to `ClayLabel` below?

Reply:
I was debating back and forth whether this should be an easier API, or if we should require the end user to wire up the appropriate close button. I sort of like having it internal to ClayLabelBase since it doesn't prohibit or get in the way at all, but I also could see moving it to just the high-level. Are you thinking we would name it `ClayLabelWithClose`? If that's the case, we could do what we did modal(withTitle) and add `withClose` as a prop that defaults to true, and then the withClose logic would ultimately get extracted to a high-level component

Reply:
Yeah, actually, I wasn't thinking of creating a `ClayLabelWithClose` now, just thinking of making `ClayLabelBase` cleaner and currently moving it to `ClayLabel` only, we will still keep supporting it. I don't think we need `withClose` here yet. We can extract it in the future when we better consolidate the ClayLabel API.